link-header-parser.lisp

```diff
@@ -0,0 +1,189 @@
+(in-package :tooter-link-header-parser)
```

Comment:
Is there a good reason for this entire parsing machinery? Can't you just use cl-ppcre and a simple regex?

Reply:
In principle i thought to use regex too but after checking the standard:

https://httpwg.org/specs/rfc8288.html#header

I was not convinced with this solution, please take a look at the BNF grammars on top of the document linked above, I found very difficult to express the grammar just with regex, so I changed my mind and  wrote a custom parser; note that this is not just a parser for link header returned by mastodon but a generic ones that can parse header with multiple parameters for each link and also parameters wrapped or not wrapped with quotation marks. Except for bugs, of course :smile: 

I think could be useful if headers become more complicated than they are now in mastodon but also other people cuould use this code for their needs, outside of mastodon API.

Bye!
C.

Reply:
I see, didn't expect the syntax to be this involved. What in the world...